### PR TITLE
Overlays

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/Dropdown.js [hunk 1]
```diff
@@ -51,10 +52,12 @@ const propTypes = {
 }
 
 const defaultProps = {
-  as: 'div',
   itemSelector: '* > *',
 }
 
+/**
```
Comment (taion) [inline src/Dropdown.js]: ?

File: src/DropdownMenu.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ class DropdownMenu extends React.Component {
      */
     flip: PropTypes.bool,
 
-    usePopper: PropTypes.oneOf([true, false, 'lazy']),
+    usePopper: PropTypes.oneOf([true, false]),
```
Comment (taion) [inline src/DropdownMenu.js]: `PropTypes.bool` then?

File: src/Overlay.js [hunk 3]
```diff
@@ -27,15 +33,36 @@ class Overlay extends React.Component {
     return null
   }
 
+  componentDidMount() {
+    this.setState({ target: this.getTarget() })
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props === prevProps) return
```
Comment (taion) [inline src/Overlay.js]: it's really ok/meaningful to do a ref equality check on all props?

shouldn't the props _always_ change?

File: src/Overlay.js [hunk 4]
```diff
@@ -48,43 +75,56 @@ class Overlay extends React.Component {
...
 
     let child = children
+    const popperProps = {
+      placement,
+      referenceElement: this.state.target,
+      enableEvents: props.show,
+      modifiers: {
+        ...modifiers,
+        preventOverflow: {
+          padding: containerPadding || 5,
```
Comment (taion) [inline src/Overlay.js]: this `5` should be a prop?

File: src/WaitForContainer.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+class WaitForContainer extends React.Component {
+  constructor(...args) {
+    super(...args)
+
+    if (!canUseDom) return
+
+    let { container } = this.props
+
+    if (typeof container === 'function') container = container()
+    if (container && !ReactDOM.findDOMNode(container)) {
```
Comment (taion) [inline src/WaitForContainer.js]: O_O i'm surprised this is safe

File: src/Overlay.js (hunk 3 above)
Comment (jquense) [reply src/Overlay.js]: It's a heuristic for "just state changed"  which is the only time they don't change

File: src/Overlay.js (hunk 4 above)
Comment (jquense) [reply src/Overlay.js]: I guess yeah, it's the popper default

File: src/WaitForContainer.js (hunk 5 above)
Comment (jquense) [reply src/WaitForContainer.js]: Yeah me too...I had it wrapped in a try/catch but no errors were ever thrown so idk

File: src/Overlay.js (hunk 4 above)
Comment (taion) [reply src/Overlay.js]: ah ok... we can't just use `undefined` then?

File: src/Dropdown.js (hunk 1 above)
Comment (jquense) [reply src/Dropdown.js]: Didn't compile otherwise, I need to add docs tho

File: src/DropdownMenu.js (hunk 2 above)
Comment (jquense) [reply src/DropdownMenu.js]: Lol

**Pull request merged**